engine/collection/epochmgr/engine.go

```diff
@@ -56,6 +57,9 @@ type Engine struct {
 	mu     sync.RWMutex                       // protects epochs map
 	epochs map[uint64]*RunningEpochComponents // epoch-scoped components per epoch
 
+	qcVoteMu          sync.Mutex                    // protects qc votes map
+	inProgressQCVotes map[string]context.CancelFunc // map of context cancel funcs, each cancel func maps to exactly 1 in progress qc voting process
+
```

Comment:
To the best of my understanding, there can only be one voting process. Therefore, a map and a mutex are unnecessarily complex, I think. Instead, I would suggest to use an atomic

```suggestion
	inProgressQCVote *atomic.Pointer[context.CancelFunc]
```

Reply:
https://github.com/onflow/flow-go/pull/5991/commits/091264c110f69ab77396c226cf45b98e9df02ba0